right/src/usb_report_updater.c

```diff
@@ -252,12 +252,54 @@ static void handleSwitchLayerAction(key_state_t *keyState, key_action_t *action)
 static uint8_t basicScancodeIndex = 0;
 static uint8_t mediaScancodeIndex = 0;
 static uint8_t systemScancodeIndex = 0;
-static uint8_t stickyModifiers, stickySlotId, stickyKeyId;
+
+/*
+ * Sticky modifiers are all "action modifiers" - i.e., modifiers of composed
+ * keystrokes whose purpose is to activate concrete shortcut. They are
+ * activated once on keydown, and reset when another key gets activated (even
+ * if the activation key is still active).
+ *
+ * Depending on configuration, they may "stick" - i.e., live longer than their
+ * activation key, either until next action, or until release of held layer.
+ * (This serves for Alt+Tab style shortcuts.)
+ */
+static uint8_t stickyModifiers;
+static uint8_t stickySlotId;
+static uint8_t stickyKeyId;
+static bool    stickyModifierShouldStick;
+
 static uint8_t secondaryRoleState = SecondaryRoleState_Released;
 static uint8_t secondaryRoleSlotId;
 static uint8_t secondaryRoleKeyId;
 static secondary_role_t secondaryRole;
 
+
+static bool shouldStick_actionIsSelected(key_action_t * action) {
+    if(action->keystroke.modifiers == 0 || action->type != KeyActionType_Keystroke || action->keystroke.keystrokeType != KeystrokeType_Basic) {
+        return false;
+    }
+
+    #define TRY_MATCH(MOD, SC) if(action->keystroke.modifiers == MOD && action->keystroke.scancode == SC) { return true; }
+
+    TRY_MATCH ( HID_KEYBOARD_MODIFIER_LEFTALT, HID_KEYBOARD_SC_TAB );
+
+    return false;
+}
+
+static bool shouldStick(key_action_t * action) {
+    bool currentLayerIsHeld = IsLayerHeld() || (secondaryRoleState == SecondaryRoleState_Triggered && IS_SECONDARY_ROLE_LAYER_SWITCHER(secondaryRole));
+    bool allowAll = true;
```

Comment:
AllowAll is the current behaviour (i.e., all modifiers stick). I view the shortcut-matching part as being the "new" code (which maybe should not be introduced yet).   

Or should this PR introduce the proposed "alt-tab-matching-only" behaviour?

Reply:
Yes, please let's only introduce the Alt+Tab matching behavior. To be more precise, let's introduce Alt+Tab and Super+Tab matching, the latter being relevant on OSX. Alt = LAlt || RAlt. Please use boolean logic instead of the proposed array solution for the sake of simplicity.

Reply:
Done, in a manner that any superset of those modifiers will match too (which I think is desirable (?)).

Reply:
Yes, well done!